### PR TITLE
Refactor custom headers to peer info

### PR DESCRIPTION
**Pull request by harbu**
Generalize `CustomHeaders` class into `PeerInfo` and keep header-specific stuff in `WsEndpoint`. Make WsEndpoint methods `send`, `sendSync`, `close` etc. operate on node-id instead of node address. These are all changes that are necessary for upcoming WebRTC branch.

### CODE REVIEW

File: src/connection/WsEndpoint.js [hunk 1]
```diff
@@ -21,58 +24,43 @@ function transformToObjectWithLowerCaseKeys(o) {
...
-    constructor(wss, customHeaders, advertisedWsUrl) {
+    constructor(wss, peerInfo, advertisedWsUrl) {
         super()
 
         if (!wss) {
             throw new Error('wss not given')
         }
-        if (!customHeaders) {
-            throw new Error('customHeaders not given')
+        if (!peerInfo) {
```
Comment (mirpo) [inline src/connection/WsEndpoint.js]: Maybe it's better instanceOf `PeerInfo` ?

File: src/connection/WsEndpoint.js [hunk 2]
```diff
@@ -213,13 +204,14 @@ class WsEndpoint extends EventEmitter {
         })
     }
 
-    onReceive(address, message) {
+    onReceive(peerInfo, address, message) {
         this.metrics.inc('onReceive')
-        this.debug('received from %s message "%s"', address, message)
-        this.emit(events.MESSAGE_RECEIVED, address, message)
+        this.debug('received from %s / %s message "%s"', peerInfo, address, message)
```
Comment (mirpo) [inline src/connection/WsEndpoint.js]: `/` or `,` ?

File: src/logic/Tracker.js [hunk 3]
```diff
@@ -18,7 +17,6 @@ module.exports = class Tracker extends EventEmitter {
 
         // set default options
         const defaultOptions = {
```
Comment (mirpo) [inline src/logic/Tracker.js]: do we need this almost empty `defaultOptions`?

File: src/protocol/NodeToNode.js [hunk 4]
```diff
@@ -59,40 +55,35 @@ class NodeToNode extends EventEmitter {
...
 
     stop() {
-        return this.basicProtocol.endpoint.stop()
+        return this.endpoint.stop()
     }
 
-    onPeerConnected(peerId) {
-        if (this.basicProtocol.peerBook.isNode(peerId)) {
-            this.emit(events.NODE_CONNECTED, peerId)
+    onPeerConnected(peerInfo) {
```
Comment (mirpo) [inline src/protocol/NodeToNode.js:73]: 🥇

**Review by mirpo — COMMENTED** [review]
What do you think about renaming all `source` to `peerId` for consistency?
For example:
```
this.protocols.trackerNode.on(TrackerNode.events.CONNECTED_TO_TRACKER, (tracker) => this.onConnectedToTracker(tracker))
        this.protocols.trackerNode.on(TrackerNode.events.TRACKER_INSTRUCTION_RECEIVED, (streamMessage) => this.onTrackerInstructionReceived(streamMessage))	        this.protocols.trackerNode.on(TrackerNode.events.TRACKER_INSTRUCTION_RECEIVED, (streamMessage) => this.onTrackerInstructionReceived(streamMessage))
        this.protocols.trackerNode.on(TrackerNode.events.TRACKER_DISCONNECTED, (tracker) => this.onTrackerDisconnected(tracker))	        this.protocols.trackerNode.on(TrackerNode.events.TRACKER_DISCONNECTED, (tracker) => this.onTrackerDisconnected(tracker))
        this.protocols.nodeToNode.on(NodeToNode.events.DATA_RECEIVED, (broadcastMessage, source) => this.onDataReceived(broadcastMessage.streamMessage, source))	        this.protocols.nodeToNode.on(NodeToNode.events.DATA_RECEIVED, (broadcastMessage, source) => this.onDataReceived(broadcastMessage.streamMessage, source))
        this.protocols.nodeToNode.on(NodeToNode.events.SUBSCRIBE_REQUEST, (subscribeMessage, source) => this.onSubscribeRequest(subscribeMessage, source))	        this.protocols.nodeToNode.on(NodeToNode.events.SUBSCRIBE_REQUEST, (subscribeMessage, source) => this.onSubscribeRequest(subscribeMessage, source))
        this.protocols.nodeToNode.on(NodeToNode.events.UNSUBSCRIBE_REQUEST, (unsubscribeMessage, source) => this.onUnsubscribeRequest(unsubscribeMessage, source))	        this.protocols.nodeToNode.on(NodeToNode.events.UNSUBSCRIBE_REQUEST, (unsubscribeMessage, source) => this.onUnsubscribeRequest(unsubscribeMessage, source))
        this.protocols.nodeToNode.on(NodeToNode.events.NODE_DISCONNECTED, (node) => this.onNodeDisconnected(node))	        this.protocols.nodeToNode.on(NodeToNode.events.NODE_DISCONNECTED, (node) => this.onNodeDisconnected(node))
        this.protocols.nodeToNode.on(NodeToNode.events.RESEND_REQUEST, (request, source) => this.requestResend(request, source))	        this.protocols.nodeToNode.on(NodeToNode.events.RESEND_REQUEST, (request, source) => this.requestResend(request, source))
```

**Review by mirpo — APPROVED** [review]
LGFM commander, only non-critical comments.

File: src/connection/WsEndpoint.js (hunk 1 above)
Comment (harbu) [reply src/connection/WsEndpoint.js]: fixed

File: src/logic/Tracker.js (hunk 3 above)
Comment (harbu) [reply src/logic/Tracker.js]: fixed

File: src/connection/WsEndpoint.js (hunk 2 above)
Comment (harbu) [reply src/connection/WsEndpoint.js]: I went with square brackets now. So:

```
received from node-2 (node) [ws://127.0.0.1:30953] message "{"version":"17.3.0","code":0,"payload":{"streams":{"stream-1::0":{"inboundNodes":[],"outboundNodes":[]}},"started":"2/10/2020, 4:27:04 PM"}}"
```

**Pull request merged**